src/hashmap.h

```diff
@@ -1,5 +1,16 @@
-#include <stddef.h>
-#include "hashmap.c"
+#ifndef HASHMAP_H_
```

Comment:
#ifndef FILENAME_H_
#define FILENAME_H_

...

#endif

This pattern is called an include guard. It prevents the same header file from being included multiple times in the same compilation unit as this would lead to errors. It doesnt matter here as the header file is only included once per compilation unit, but it is standard to always have this anyways

Reply:
I thought `#pragma once` was doing the same thing?

Reply:
Yeah it does, but anything starting with #pragma is actually compiler specific although #pragma once works on all the popular C compilers. Although some companies still uses #ifndef bc of this e.g. Google